support/langkit_support-symbols.ads

```diff
@@ -29,29 +29,63 @@ with GNAT.String_Hash;
 with Langkit_Support.Text; use Langkit_Support.Text;
 with Langkit_Support.Vectors;
 
---  Provide a symbol table for text (Text_Type) identifiers
+--  This package provides a symbol table for text (Text_Type) identifiers. This
+--  is used in Langkit for the interning of symbols.
+--
+--  The main interest is to:
+--
+--  1. Use less memory by interning strings
+--
+--  2. Be faster when using the symbol as map keys for example, because hashing
+--     is faster.
 
 package Langkit_Support.Symbols is
 
    type Symbol_Table_Record is tagged private;
+
    type Symbol_Table is access all Symbol_Table_Record'Class;
-   --  Represents a symbol table
+   --  Represents a symbol table. The symbol table is the holder of the memory
+   --  allocated for each symbol, and serves as a single access point if you
+   --  want to find back an existing symbol.
 
    No_Symbol_Table : constant Symbol_Table;
    --  Value to use as a default for unallocated symbol tables
 
    type Symbol_Type is new Text_Cst_Access;
+   --  Main symbol type.
+   --
+   --  WARNING: For usability reasons, we use the access to the string as a
+   --  symbol type. This is very convenient because you can access the text of
+   --  a symbol without a reference to the symbol table, but is also unsafe,
+   --  because if the symbol table has been freed, the symbol will be a
+   --  dangling pointer.
 
    function Image (S : Symbol_Type) return Text_Type;
+   --  Return the text associated to this symbol.
```

Comment:
FYI: single-line Ada comments should not end with a dot… I’m surprised that the style checker hasn’t caught this (there are multiple instances in this patch). Oh well, I’ll have a look later at the checker to see what’s going on.